mxcube3/routes/signals.py

```diff
@@ -121,6 +121,15 @@ def sc_unload(location):
 
     socketio.emit('sc', msg, namespace='/hwr')
 
+def path_safe_changed(*args):
+    # responds to the 'pathSafeChanged' signal in cats like SC
+    new_state = args[0]
+    #we are only interested when it becames true
+    if new_state:
```

Comment:
Wont it always remain true then ?, or is it re-setted somewhere else ?

Reply:
it will switch between false/true. We only want to react when is safe, so that the loading panel disappears. The signal comes from the SC

Reply:
I see, but what hides the dialog ?

Reply:
Okey, but what if it for some reason goes false after that ?

Reply:
To clarify a bit, I guess its not a big deal in the end. But imagine this case loading is false and "is safe" turns false, what would happen then ? Not sending the when state is false would prevent us from handling such cases. They are probably rare so its maybe not a big deal ?

Reply:
Okey, well lets leave it like this for the moment then. I still think that it's a good idea to consider situations when we are not loading but are "not safe" but for now its fine then.